File: order_service/src/main/java/com/huike/order/OrderApplication.java
package com.huike.order;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.cloud.client.loadbalancer.LoadBalanced;
import org.springframework.context.annotation.Bean;
import org.springframework.web.client.RestTemplate;

@SpringBootApplication
@EntityScan("com.huike.order.entity")

public class OrderApplication {
    @LoadBalanced
    @Bean
    public RestTemplate restTemplate(){
        return  new RestTemplate();
    }

    public static void main(String[] args) {
        SpringApplication.run(OrderApplication.class,args);
    }
}

File: order_sertvice/src/main/java/com/huike/order/OrderApplication.kt
package com.huike.order

import com.huike.order.startlintener.SpringStartListener
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.domain.EntityScan
import org.springframework.context.ConfigurableApplicationContext
import org.springframework.context.annotation.Bean
import org.springframework.web.client.RestTemplate

@SpringBootApplication
@EntityScan("com.huike.order.entity")
open class OrderApplication {

    @Bean
    open fun restTemplate(): RestTemplate {
        return RestTemplate()
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            println("Hello world!")
            val configurableApplicationContext: ConfigurableApplicationContext = SpringApplication.run(OrderApplication::class.java, *args)
            // 向Spring上下文中添加监听器
            configurableApplicationContext.addApplicationListener(SpringStartListener())
        }
    }
}
